### ISSUE TITLE
Training only on KITTI 

### ISSUE THREAD

**Issue by OzzyW** [lmb-freiburg/dispnet-flownet-docker#7]
Hey Nikolaus,

So here is a sample of the mid results after 600K iterations for DispNet:

1) Example of a image from testing (no gt available) for only KITTI training -
![kitti3](https://user-images.githubusercontent.com/21277219/40916010-6bb4022e-67fe-11e8-849a-8e73bc6ca830.png)
2) Example of a image from testing (no gt available) for your original Dataset -
![nor3](https://user-images.githubusercontent.com/21277219/40916007-68b20c9c-67fe-11e8-917c-aeffd0149ee8.png)

Did you use some caffe visualization tools to set the max iterations to 1200K to know the net was totally converged (or was overfitting) or just by the checking the loss output? Also, is comparing the testing images (with no ground truth provided) to original results the best way to prove if there is actually overfitting here?

**Comment by nikolausmayer** [comment]
Hi,

I think we relied on "looking at the loss", coupled perhaps with "how many more iterations until the deadline?" ;)

Anyway, that's an interesting result. I would have guessed that the network learns that it can output anything at depth boundaries, but it looks like the network actually learns something more useful. What are those dark areas around the posts? Near zero?

It's also clear that the KITTI network has never seen data for the upper region of the image :)

**Comment by OzzyW** [comment]
Great answer haha... It is cool that it really tries to imitate the gt which is kinda also semi-synthetic.
Are you referring to the black regions around the thin structures?

**Comment by nikolausmayer** [comment]
Yes! The black regions exist in the ground truth because the laser scanner is bad at glancing angles and because of the perspective difference between the scanner and the camera (if I understand it correctly). 
However, if we finetune a DispNet that's pretrained on our synthetic data, then the network learns that it can predict whatever it wants in these regions and in the sky. It does _not_ learn to produce a "black" sky, so I'm wondering what the dark regions in your KITTI-trained result are. Are they really small disparities? At first glance it looks like you treated missing KITTI pixels as "zero" and the network learned this?

**Comment by OzzyW** [comment]
let me understand how to manipulate the output (for some reason I got the DispNetCorr1D-K output to work for me but not the DispNet) and will check exactly what are the values of some black interest points. I actually did not yet preprocess anything. Just wanted see what kind of results I get with the original KITTI gt.

**Comment by OzzyW** [comment]
let me understand how to manipulate the output and will check exactly what are the values of some black interest points. I might have done something like what you are describing inadvertently when converting files to PFM? Anyways, the network did seem to lean it!

**Comment by OzzyW** [comment]
So although it learned that the sky and the black region next to thin objects is closer to zero the problem is I am not sure what this can tell us since the scale was wrong. In addition I treated missing KITTI pixels as "zero" and not NaN. 

What do you think @nikolausmayer? 
Also, can it be integrated to gain any benefit with the normal fine-tuning or just an interesting experiment?
 
Finally, this seems to solve the scaling issue:
https://github.com/lmb-freiburg/flownet2/issues/40

**Comment by nikolausmayer** [comment]
I think a wrong scale on the whole image does not invalidate such an experiment. Learning a constant scale is easy for a network this deep.
I would count training-with-zeroes as "interesting experiment", although some people might say that "it's obvious that the network would learn this". To me this shows that the network learns context information, something like "there must be zero-regions at depth discontinuities".

About benefits, I'm not sure. A network like this could be used to generate KITTI-like disparity maps for street scenes, and maybe there's some value in that if it can be used e.g. in the context of style transfer. But since it's effectively learning a wrong modality (= not what you actually want from a disparity network) it might be hard to sell it on its own :wink:

**Comment by OzzyW** [comment]
I guess for me nothing is obvious. :nerd_face: 

BTW you have some insight about what **not to** try to improve this network (might there be more redundency with some layers that only trial and error will find like this https://github.com/lmb-freiburg/flownet2/issues/54)? 

The last paper I saw is "What Makes Good Synthetic Training Data for Learning
Disparity and Optical Flow Estimation?" Anything like dispnet 2.0 coming out soon (as I am a fan already)?

**Comment by nikolausmayer** [comment]
Nah, nothing is obvious until you have tried it :grin: and hindsight is 20/20

Hm. "Don't reduce the number of channels of the whole network", maybe.

We don't really know how many convolutional layers are needed except for the up/downsampling layers. We've done a few experiments with Batch Normalization, but nothing too conclusive. We've experimented with shortcut configurations, but not a lot. And there are currently some papers about alternative/extended layers, like [Shift](https://arxiv.org/abs/1711.08141) or [CoordConv](https://arxiv.org/abs/1807.03247). So many possible experiments!

About DispNet 2.0: keep your eyes peeled. I cannot comment :wink:

**Comment by OzzyW** [comment]
Cool, I'll keep my eyes :eyes: open.
 Thank you very much!!

**Comment by nikolausmayer** [comment]
I guess now it's fine for me to advertise this: the next-level DispNet will be presented at ECCV 2018; here's the paper: https://lmb.informatik.uni-freiburg.de/Publications/2018/ISKB18/

**Comment by OzzyW** [comment]
Oh very interesting, I also thought the network seems to be very generic. 
Too bad there are no more tickets to ECCV to talk you guys there :smile: 
Thanks for the update.

BTW, for the sake of transparency, I think that the Just KITTI training I did before was on the "wrong" data here so there are multiple problems with this experiment.

**Comment by nikolausmayer** [comment]
:grin:  aah, the good old "wrong data" problem. At least it keeps this issue lively!

**Comment by OzzyW** [comment]
I did not know it was so common :laughing:

**Comment by nikolausmayer** [comment]
(closed due to inactivity)
